dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/h2internal/H2StreamProcessor.java

```diff
@@ -1952,34 +1948,37 @@ private boolean writeFrameSync() throws FlowControlException, Http2Exception {
         if (currentFrame.isWriteFrame() && currentFrame.getInitialized()) {
             WsByteBuffer writeFrameBuffer = null;
             WsByteBuffer[] writeFrameBuffers = null;
+            int writeTimeout = muxLink.config.getWriteTimeout();
             try {
                 if (currentFrame.getFrameType() == FrameTypes.DATA) {
                     FrameData data = (FrameData) currentFrame;
                     boolean timedOut = false;
 
-                    // Check to see if the write window is large enough to write this data.
+                    // Check to see if the write window is large enough to write this data
+                    // if it's not, wait for at most the configured writeTimeout period (default 60s)
                     if (isWindowLimitExceeded((FrameData) currentFrame)) {
-                        // the connection or stream window is too small to write this data frame.  This thread will wait for a max of
-                        // 5 seconds for a window update that's large enough to allow the data frame to be written out
                         long startTime = System.currentTimeMillis();
                         while (isWindowLimitExceeded((FrameData) currentFrame) && !timedOut) {
                             synchronized (this) {
-                                this.wait(MAX_TIME_TO_WAIT_FOR_WINDOW_UPDATE_MS);
+                                this.wait(writeTimeout);
                             }
+                            long elapsed = System.currentTimeMillis() - startTime;
```

Comment:
Can we declare "elapsed" outside of the while loop so that a new object isn't created on every loop iteration?